internal/pkg/pluginengine/change_helper.go

```diff
@@ -42,6 +43,29 @@ func generateAction(tool *configmanager.Tool, action statemanager.ComponentActio
 	}
 }
 
+func ResourceDrifted(fromStateFile, fromRead map[string]interface{}) (bool, error) {
+	// nil vs empty map
+	if cmp.Equal(fromStateFile, fromRead, cmpopts.EquateEmpty()) {
+		return false, nil
+	}
+
+	// use marshal and unmarshal to remove type details
+	fromReadBytes, err := yaml.Marshal(fromRead)
+	if err != nil {
+		return true, err
+	}
+	fromReadWithoutType := map[string]interface{}{}
+	err = yaml.Unmarshal(fromReadBytes, &fromReadWithoutType)
+	if err != nil {
+		return true, err
+	}
+	log.Info(fromStateFile)
+	log.Info(fromReadWithoutType)
```

Comment:
Will these two printing with info level cause too much output?

Reply:
> Will these two printing with info level cause too much output?

Sorry, it is for the test. I forget to delete them. You are very careful!